common/djangoapps/enrollment/tests/test_api.py

```diff
@@ -56,6 +58,35 @@ def test_enroll(self, course_modes, mode):
         get_result = api.get_enrollment(self.USERNAME, self.COURSE_ID)
         self.assertEquals(result, get_result)
 
+    @ddt.data(
+        ([CourseMode.DEFAULT_MODE_SLUG, 'verified', 'credit'], CourseMode.DEFAULT_MODE_SLUG),
+        (['audit', 'verified', 'credit'], 'audit'),
+        (['honor', 'verified', 'credit'], 'honor'),
+    )
+    @ddt.unpack
+    def test_enroll_no_mode_success(self, course_modes, expected_mode):
+        # Add a fake course enrollment information to the fake data API
+        fake_data_api.add_course(self.COURSE_ID, course_modes=course_modes)
+        # Enroll in the course and verify the URL we get sent to
+        result = api.add_enrollment(self.USERNAME, self.COURSE_ID)
+        self.assertIsNotNone(result)
+        self.assertEquals(result['student'], self.USERNAME)
+        self.assertEquals(result['course']['course_id'], self.COURSE_ID)
+        self.assertEquals(result['mode'], expected_mode)
+
+    @ddt.data(
+        [],
+        ['professional'],
+        ['verified'],
+        ['verified', 'professional'],
+    )
+    @raises(CourseModeNotFoundError)
+    def test_enroll_no_mode_error(self, course_modes):
+        # Add a fake course enrollment information to the fake data API
+        fake_data_api.add_course(self.COURSE_ID, course_modes=course_modes)
+        # Enroll in the course and verify the URL we get sent to
+        api.add_enrollment(self.USERNAME, self.COURSE_ID)
```

Comment:
I think the comment above this needs to be updated, and this test could use a comment explaining that we're testing that `CourseModeNotFoundError` is thrown when trying to enroll without specifying a mode when the course doesn't have a defaultable mode (i.e. `honor` or `audit`).